### PR TITLE
randomness #2: real dkg from randomnet

### PR DESCRIPTION
**Pull request by zjma**

### CODE REVIEW

File: types/src/dkg/real_dkg/mod.rs [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+pub fn build_dkg_pvss_config(
+    cur_epoch: u64,
+    next_validators: &[ValidatorConsensusInfo],
+) -> DKGPvssConfig {
+    let validator_stakes: Vec<u64> = next_validators.iter().map(|vi| vi.voting_power).collect();
+
+    let dkg_rounding =
+        DKGRounding::new(&validator_stakes, SECRECY_THRESHOLD, RECONSTRUCT_THRESHOLD);
+
+    println!(
```
Comment (danielxiangzl) [inline types/src/dkg/real_dkg/mod.rs:60]: We should move some parts to other places so we can properly add logs / alert.

File: types/src/dkg/real_dkg/mod.rs [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+            &pub_params.pvss_config.wconfig,
+            &Player {
+                id: player_idx as usize,
+            },
+            dk,
+        );
+        Ok((sk, pk))
+    }
+
+    fn reconstruct_secret_from_shares(
```
Comment (danielxiangzl) [inline types/src/dkg/real_dkg/mod.rs:223]: This should only be used for testing?

**Review by danielxiangzl — APPROVED** [review]

File: types/src/dkg/real_dkg/mod.rs [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+
+    let wconfig = dkg_rounding.wconfig.clone();
+
+    let pp = DkgPP::default_with_bls_base();
+
+    DKGPvssConfig::new(cur_epoch, wconfig.clone(), pp, consensus_keys)
+}
+
+#[derive(Debug)]
+pub struct RealDKG {}
```
Comment (zekun000) [inline types/src/dkg/real_dkg/mod.rs:83]: let's not call it RealDKG, how about wDKG or something?

**Review by zekun000 — APPROVED** [review]

File: types/src/dkg/real_dkg/mod.rs (hunk 3 above)
Comment (zjma) [reply types/src/dkg/real_dkg/mod.rs:83]: will rename later!

**Pull request merged**